swan-lake/learn-the-platform/test-document-the-code/debug-ballerina-programs.md

```diff
@@ -7,262 +7,22 @@ permalink: /learn/debug-ballerina-programs/
 active: debug-ballerina-programs
 ---
 
-When writing large-scale applications with complex logic, it is quite possible to have syntax, semantic, 
-or runtime issues in your code. The Ballerina compiler comes in handy when detecting syntax and semantic issues. 
-However, it is very difficult for a compiler to detect runtime errors like logical errors because they occur during the 
-program execution after a successful compilation. This is where the dedicated debugging tooling support becomes important.
+The Ballerina compiler comes in handy to detect syntax and semantic issues that occur in your code when developing large-scale applications with complex logic. However, it is difficult for the compiler to detect runtime errors like logical errors because they occur during the program execution after a successful compilation. This is where the dedicated debugging tooling support of Ballerina becomes important.
 
-Currently, we provide a feature-rich debugging experience for troubleshooting Ballerina applications
-via the <a href="https://marketplace.visualstudio.com/items?itemName=WSO2.Ballerina" target="_blank">Ballerina Visual Studio Code extension</a>.
+## Debugging overview
 
-## Start a debug session
+The Ballerina VS Code extension offers three types of sessions (i.e., program, test, and remote). Debugging can be done using CodeLens or configurations in a `launch.json file`. It also provides a **Debug Console** to view the output and perform various debugging scenarios.
 
-The Ballerina extension provides multiple options to debug Ballerina applications, and the most convenient way will be using the context-aware debug codelens support.
+>**Info:** For more information on the debugging sessions and methods, go to <a href="https://wso2.com/ballerina/vscode/docs/debug-the-code/debugging-overview/" target="_blank">Debug sessions</a>.
 
-However, if you require launching debug sessions with additional configurations (i.e., program arguments, environment variables), you can use `launch.json` configurations
-(For more information, see [Start a debug session with configurations](#start-a-debug-session-with-configurations)).
+## Debugging configurations
 
-### Start a program debug session
+You can define configuration attributes in the `launch.json` file to apply them on the above debug sessions.
```

Comment:
```suggestion
The Ballerina debugger allows you to create a `launch.json` file with default configurations for debugging Ballerina programs. By following a few steps, you can generate the file and then modify the configurations to suit your needs. These configurations have specific attributes that can be edited to customize the debugging process, so that you can set program arguments, command options, environment variables, run in a separate terminal, etc. Additionally, you can configure remote debugging by specifying the host address and port number. These configurations provide flexibility and control when debugging Ballerina code in VS Code.
```